wallyworld/juju
# Cache discharge macaroons in cross model relations client

## Description of change

Add a new expiring cache to hold macaroons. This is used by the cross model relations client so that any discharge macaroons for one api call can be reused with the next (until the macaroons expire and new ones are obtained). This avoids the need to discharge all the time.

## QA steps

Run up a cmr scenario and ensure everything works as before - the change should be transparent.
